src/components/modals/confirmationDialog.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <Teleport to="#modals">
```

Comment:
shouldn't Teleport live in baseModal?

Reply:
Doing something not clean (since you have to use teleport inside every component that will use baseModal...) to make tests work isn't right.

Since you don't want to test here if teleport works why you don't just disable it using vite test stubs like explained [here](https://test-utils.vuejs.org/guide/advanced/teleport.html#Testing-Teleport ) in the TIP section ? This would also simplify the test since you don't have to use any strange before/after hook, and `wrapper.getComponent` on baseModal.

```ts
import { mount } from '@vue/test-utils';
import confirmationDialog from './confirmationDialog.vue';

test('confirmationDialog', async () => {
  const wrapper = mount(confirmationDialog, {
    props: { show: true },
    global: {
      stubs: {
        teleport: true,
      },
    },
  });

  await wrapper.get('#modal-confirm').trigger('click');
  expect(wrapper.emitted()).toHaveProperty('result');
  const result = wrapper.emitted('result')!;
  expect(result).toHaveLength(1);
  expect(result[0]).toStrictEqual([true]);
});
```

Reply:
@sgotti thank you for the solution. I fixed it.